### PR TITLE
Fix Readme render bug

### LINKED ISSUE #19988: README rendering broken
https://try.gitea.io/wxiaoguang/test

image

### PR DESCRIPTION
**Pull request by lunny**
Fix #19988

The problem is caused sometiems `TreePath` is not the file path but the file's parent dir path.

This is a temporory bug fix, the same variable should not have different ,meaning.

For README render, it's an index file, so `treepath` is `""` or `"/"`, but we need to render the `README.md`. That's the problem.

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

File: routers/web/repo/view.go [hunk 1]
```diff
@@ -357,7 +357,7 @@ func renderReadmeFile(ctx *context.Context, readmeFile *namedBlob, readmeTreelin
 		var result strings.Builder
 		err := markup.Render(&markup.RenderContext{
 			Ctx:          ctx,
-			RelativePath: ctx.Repo.TreePath,
+			RelativePath: path.Join(ctx.Repo.TreePath, readmeFile.name), // For README renderer, ctx.Repo.TreePath just the path without file name
```
Comment (zeripath) [inline routers/web/repo/view.go]:
```suggestion
			RelativePath: path.Join(ctx.Repo.TreePath, readmeFile.name), // ctx.Repo.TreePath is the directory not the Readme so we must append the Readme filename (and path).
```

**Comment by zeripath** [comment]
I'm not sure that there's actually any overloading of the TreePath going on here.

The TreePath is the path that the user has requested. They didn't request the README file explicitly - we're just additionally rendering the README file within the rendering of the directory.

**Review by zeripath — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> I'm not sure that there's actually any overloading of the TreePath going on here.
> 
> The TreePath is the path that the user has requested. They didn't request the README file explicitly - we're just additionally rendering the README file within the rendering of the directory.

How about a README in a subdir ?

https://try.gitea.io/wxiaoguang/test/src/branch/master/subdir
https://try.gitea.io/wxiaoguang/test/src/branch/master/subdir/README.md

**Review by zeripath — DISMISSED** [review]
~~This won't work for readmes that are actually within docs/ .gitea/ .github/~~

Ignore me readmeFile.name has the right path bits.

**Comment by wxiaoguang** [comment]
> ~This won't work for readmes that are actually within docs/ .gitea/ .github/~
> 
> readmeFile.name has the right path bits.

But it works, I have tested.

image

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
> > ~This won't work for readmes that are actually within docs/ .gitea/ .github/~
> > readmeFile.name has the right path bits.
> 
> But it works, I have tested.

The test you were describing was subdir/ readmes. I was actually concerned about the case where TreePath is "" but the readme file is in `docs/README.md`, `.github/README.md` or `.gitea/README.md`.

In those cases, readmeFile.Name actually is as above - in order that this kind of thing worked transparently and I had forgotten that that was the case.

**Comment by wxiaoguang** [comment]
Sorry for that I didn't say it clearly .....

**Pull request merged**